File: medium/cpp/1167.cpp
// https://leetcode.com/problems/minimum-cost-to-connect-sticks/

class Solution
{
public:
    int connectSticks(std::vector<int>& sticks)
    {
        std::priority_queue<int, std::vector<int>, std::greater<>> heap(std::begin(sticks), std::end(sticks));

        auto res = 0;

        while (std::size(heap) > 1)
        {
            const auto first = heap.top();
            heap.pop();

            const auto second = heap.top();
            heap.pop();

            heap.push(first + second);
            res += first + second;
        }

        return res;
    }
};

File: medium/python/1167.py
// https://leetcode.com/problems/minimum-cost-to-connect-sticks/description/

class Solution:
    def connectSticks(self, sticks: List[int]) -> int:
        heapify(sticks)
        n = len(sticks)
        res = 0

        while n > 1:
            n -= 1
            first, second = heappop(sticks), heappop(sticks)
            heappush(sticks, first + second)
            res += first + second

        return res
        